### DISCUSSION TITLE
Formula for Terrain

### DISCUSSION THREAD

**Discussion by MeusRex** [MeusRex/pf2e-chex#9] [Ideas]
It would make sense that the terrain can set some values on a hex.
Example: Forest might set Food 1 - 3.
E.g. formulas should make it possible to random seed hexes.

The question is how? Connect a roll Table? Or do everything with formulas like travel?

**Comment by federicoacker** [comment]
I feel like the rollable table while being more work to implement for GMs has the most flexibility on what it would allow users to do, at the same time it might be better to at first offer a solution like with improvements, where we can type in a formula (maybe using foundry's macro syntax) to add resources. Also I don't really know where to put this as a suggestion, but it would be nice to make it so that Features that are added to a tile could be used to automatically add foregrables or resources to said tile, so for example, I created a feature called "Wyvern's nest" that is supposed to indicate to my players that the tile is a dangerous area but at the same time automatically add food and ore as a forageable/resource to the tile so I can go around adding those features where I need them and that would in general streamline the process

**Reply by federicoacker** [reply]
Actually scratch that, foundry's rollable tables are way too clunky and require too much setup, treating it like you did with the travel part is honestly perfect

**Comment by MeusRex** [comment]
Hmm I'll have to think about this. In theory I could make features "seedable". You select a terrain, set a percent chance and then when you set terrain it could get seeded.
But it needs some consideration. How would the UI look. It needs to be kinda self explanatory, I don't really want to make a video :D
Also, would painting terrain replace the existing features? If a hex is repainted a bunch of times, you could get a collection of wierd features on it otherwise.

I was also thinking that I was going to add a feature/improvement painter. It would be an option to add a randomize there, instead of linking it to the terrain.

**Comment by federicoacker** [comment]
Mhhhhh, in that sense definitely I think when you repaint a hex it should first clear it of everything it has and then reseed it with the new stuff. When it comes to feature painting maybe, something similar to the terrain palette could work UI-wise while for improvements given that they are like buildings maybe having some kind of window open up with a list of improvements, that maybe we could divide by resource type they add, and then just clicking on the improvement you want and clicking on the tile you want it on

**Reply by federicoacker** [reply]
to add to this, you could make us of the foundry file manager system so that when an improvement is added to a tile a custom image linked to that improvement gets shown over the tile in the chex overlay? Not sure how hard this would be to implement, especially when it comes to auto scaling the image and moving it in the case that multiple improvements are added, this might need you to implement a hard-cap on how many improvements are added or a soft-cap where after say 4 improvements no more images are added on the tile and the improvements just show up in the window that opens when you hover with your mouse
